tendermint/Cargo.toml

```diff
@@ -51,14 +52,14 @@ tendermint-proto = { version = "0.25.0", default-features = false, path = "../pr
 time = { version = "0.3", default-features = false, features = ["macros", "parsing"] }
 zeroize = { version = "1.1", default-features = false, features = ["zeroize_derive", "alloc"] }
 flex-error = { version = "0.4.4", default-features = false }
-k256 = { version = "0.11", optional = true, default-features = false, features = ["ecdsa", "sha256"] }
+k256 = { version = "0.11", default-features = false, features = ["ecdsa", "sha256"] }
```

Comment:
Having this one optional depends on whether `CryptoProvider` will be under a feature flag.

Reply:
As I understand it now, we lock the signature type in `CryptoProvider` to the one defined by `k256`. If we cannot abstract this associated type to less specific rust-crypto bounds ("must be fixed array of 256 bits"?) and still keep it usable in generic code, then `k256` should become a hard dependency. But I thought dependency management was the whole purpose of this rework.